src/Adapter/Chain/CachePoolChain.php

```diff
@@ -111,8 +152,15 @@ public function hasItem($key)
     public function clear()
     {
         $result = true;
-        foreach ($this->pools as $pool) {
-            $result = $result && $pool->clear();
+        foreach ($this->getPools() as $poolKey => $pool) {
+            try {
+                $result = $result && $pool->clear();
+            } catch (\Exception $e) {
+                if (!$this->skipOnFailure) {
+                    throw $e;
+                }
+                unset($this->pools[$poolKey]);
+            }
```

Comment:
These 9 lines of code is (almost) duplicated 10 times in this class. Is it possible to do something smart about it? Something that also would increase readability.